test/filters/test_clipper.cpp

```diff
@@ -287,7 +303,172 @@ TEST (CropBox, Filters)
   cropBoxFilter.filter (indices);
   EXPECT_EQ (int (indices.size ()), 9);
 
-  // PCLPointCloud2
+  // PointCloud with indices selection
+  // -------------------------------------------------------------------------
+
+  // Reset cropBox transformation 
+  cropBoxFilter.setNegative (false);
+  cropBoxFilter.setTransform (getTransformation(0, 0, 0, 0, 0, 0));
+  cropBoxFilter.setRotation (Eigen::Vector3f (0.0f, 0.0f, 0.0f));
+  cropBoxFilter.setTranslation (Eigen::Vector3f (0.0f, 0.0f, 0.0f));
+
+  // Setup input indices selection
+  cropBoxFilter.setIndices(idx);
+
+  // Indices
+  cropBoxFilter.filter (indices);
+
+  // Cloud
+  cropBoxFilter.filter (cloud_out);
+
+  // Should contain all
+  EXPECT_EQ (int (indices.size ()), 7);
+  EXPECT_VECTOR_CONTAINS_ALL (std::vector<int>({1, 2, 3, 5, 6, 7, 8}), indices);
```

Comment:
Is it possible to write `{1, 2, 3, 5, 6, 7, 8}` here, does the compiler manage to deduct type? Would be cleaner.

Reply:
Unfortunately, MACRO doesn't figure out initializer-list and uses each element as a single parameter. I'm not really comfortable with variadic macro, I found some partial solutions but they don't cover all situations.